include/flatbuffers/util.h

```diff
@@ -45,13 +50,65 @@
 
 namespace flatbuffers {
 
+#ifdef FLATBUFFERS_PREFER_PRINTF
+template<typename T> size_t IntToDigitCount(T t) {
```

Comment:
I agree it's a bit more verbose. This does allocate less memory (only what is required, plus 1 for the extra null byte). I can remove this calculation and allocate the maximum amount for a double (387 chars, IIRC) then resize down, but that seemed like a lot of overhead which would rarely be used. Whatever is preferred, though, I agree it could be a much smaller change without this.

Reply:
(there is a portable way to get that 387 number BTW, that's just my recollection for the approximate worst-case char count)

Reply:
Yes, that be fine, especially since it is not the default path.
Might as well write it to a char buffer, then create the string from its contents in the end rather than resizing (that way the result is small when it can be).